src/dataspaces.jl

```diff
@@ -78,13 +78,17 @@ dataspace(n::Nothing) = Dataspace(API.h5s_create(API.H5S_NULL))
 # for giving sizes explicitly
 """
     dataspace(dims::Tuple; max_dims::Tuple=dims)
+    dataspace(dims::Tuple, max_dims::Tuple)
 
 Construct a simple `Dataspace` for the given dimensions `dims`. The maximum
 dimensions `maxdims` specifies the maximum possible size: `-1` can be used to
 indicate unlimited dimensions.
 """
 dataspace(sz::Dims{N}; max_dims::Union{Dims{N},Tuple{}}=()) where {N} =
     _dataspace(sz, max_dims)
+dataspace(sz::Dims{N}, max_dims::Union{Dims{N},Tuple{}}) where {N} =
+    _dataspace(sz, max_dims)
+dataspace(dims::Tuple{Dims{N},Dims{N}}) where {N} = _dataspace(first(dims), last(dims))
```

Comment:
should we document these additional methods? These  were added for additional flexibility?

Reply:
I think that we just implemented what wa already documented per #1084, but yes the documentation should reflect the new methods.

Reply:
I see the fix for adding the methods for `create_dataset` as per documentation, I was just curious on the new `dataspace` methods.